plugins/WireframeView/WireframeView.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2017 Tim Kuipers
+# Cura is released under the terms of the AGPLv3 or higher.
+
+import math
+import os.path
+
+from UM.Application import Application
+from UM.Math.Color import Color
+from UM.PluginRegistry import PluginRegistry
+from UM.Preferences import Preferences
+from UM.View.View import View
+from UM.Scene.Iterator.BreadthFirstIterator import BreadthFirstIterator
+
+from UM.View.RenderBatch import RenderBatch
+from UM.View.GL.OpenGL import OpenGL
+
+from UM.Settings.Validator import ValidatorState
+from cura.Settings.ExtruderManager import ExtruderManager
+
+
+## View used to display the edges of objects.
+class WireframeView(View):
+    def __init__(self):
+        super().__init__()
+
+        self._wireframe_shader = None
+
+    def beginRendering(self):
+        scene = self.getController().getScene()
+        renderer = self.getRenderer()
+
+        if not self._wireframe_shader:
+            self._wireframe_shader = OpenGL.getInstance().createShaderProgram(os.path.join(PluginRegistry.getInstance().getPluginPath("WireframeView"), "wireframe.shader"))
+            self._wireframe_shader.setUniformValue("u_color", Color(*Application.getInstance().getTheme().getColor("xray").getRgbF()))
```

Comment:
This is not very semantically correct, is it? This has nothing to do with "xray".